JS/removeItem.js

```diff
@@ -0,0 +1,33 @@
+
+function removeInCart(arr, id) {
+    const myArr = [...arr];
+    let idx = myArr.findIndex((e) => {
+        if (e.id == id)
+            return true;
+    });
+    if (idx == -1) return;
+    myArr.splice(idx, 1);
+    return myArr;
+}
+
+const DeleteProduct = (data, id) => {
+    let idx = data.findIndex((e) => {
+        if (e.id == id)
+            return true;
+    });
+    if (idx == -1) return 0;
+    data.splice(idx, 1);
+    return data;
+}
+function AddToArray(array , obj){
+    if(!array) array = []
+    array.push(obj)
+    return array
```

Comment:
here you are updating the original array(that is passed as an input to the function) by pushing a new element to it which is not pure. 
```js
const addNewProduct = (products, product) => [...products, product]
```